plugins/sql_config_validator.h

```diff
@@ -98,6 +111,14 @@ template<typename FieldType> struct NoGroupByHaving
     }
 };
 
+template<typename FieldType> struct NoGroupByHaving<Optional<FieldType> >
```

Comment:
Not sure if I fully understand the question.  The second template is a partial template specialization.  It is preferred whenever we instantiate the template NoGroupByHaving with an optional type.  This allows for testing if the optional is set and if so do the rest of the validation.